src/engine/sparqlExpressions/SparqlExpressionValueGetters.h

```diff
@@ -118,6 +120,53 @@ struct StringValueGetter {
 
   std::optional<string> operator()(IdOrString s,
                                    const EvaluationContext* ctx) const {
+    return std::visit(
+        [self = this, ctx](auto el) { return self->operator()(el, ctx); },
+        std::move(s));
+  }
+};
+using StringValueGetter = StringValueGetterImpl<true>;
+using StringValueGetterRaw = StringValueGetterImpl<false>;
+
+// Value getters for `isIRI`, `isBlank`, and `isLiteral`.
+template <auto isSomethingFunction, auto prefix>
+struct IsSomethingValueGetter {
+  Id operator()(ValueId id, const EvaluationContext* context) const {
+    return Id::makeFromBool(id.getDatatype() == Datatype::VocabIndex &&
+                            std::invoke(isSomethingFunction,
+                                        context->_qec.getIndex().getVocab(),
+                                        id.getVocabIndex()));
+  }
+  Id operator()(const std::string& s, const EvaluationContext*) const {
+    return Id::makeFromBool(s.starts_with(prefix));
+  }
+  Id operator()(IdOrString s, const EvaluationContext* ctx) const {
+    return std::visit([this, ctx](auto el) { return operator()(el, ctx); },
+                      std::move(s));
```

Comment:
Unfortunately you currently have to do the whole `self = this` shenanigans as in the other value getters to silence the false positive warning in clang-17.

I will think of something that abstracts away those redundant overloads for IdOrString soon anyway.